test/Intent/Runner.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Intent;
+
+using System.Reflection;
+
+public class Runner
+{
+    public static void Run(IEnumerable<string> path, IRunLogger logger)
+    {
+        foreach (var p in path)
+        {
+            try
+            {
+                var asm = Assembly.LoadFrom(p);
+                if (asm.IsExcluded())
+                    continue;
+
+                var ts = asm.GetTypes().SkipNonPublic().SkipExcluded();
+                foreach (var t in ts)
+                {
+                    var ms = t.GetMethods().SkipExcluded();
+
+                    // TODO: This chooses the Only tests only for single assembly and single class,
+                    // to support this full we would have to enumerate all classes and methods first,
+                    // it is easy, I just don't need it right now.
+                    var only = ms.Where(m => m.GetCustomAttribute<OnlyAttribute>() != null).ToList();
+                    if (only.Any())
```

Comment:
`Any` always causes evaluation but we have a list so the `Count` property is already processed.

```suggestion
                    if (only.Count > 0)
```

Reply:
done. FWIW probably makes no noticeable difference, especially considering we will have 0-2 methods in that list :)